registry/spec.md

```diff
@@ -1,57 +1,289 @@
+- Description: A mechanism in which additional metadata about the entity can
+  be stored without changing the schema of the entity.
+- Constraints:
+  - if present, MUST be a map of zero or more name/value string pairs
+  - each name MUST be a non-empty string consisting of only alphanumeric
+    characters, `-`, `_` or a `.`; be no longer than 63 characters;
+	start with an alphanumeric character and be unique within the scope of
+	this map. Values MAY be empty strings
+- Examples:
+  - `{ "owner": "John", "verified": "" }`
+
+### `version`
+- Type: Unsigned Integer       # SHOULD this be a String?
+- Description: A numeric value representing a specific instance of an entity.
+  Note that versions of an entity can be modified without changing the
+  `version` value. Often this value, or when a new version is created, is
+  controlled by a user of the Registry. Also see `epoch`.
+- Constraints:
+  - MUST be an unsigned integer equal to or greater than zero
+  - older versions of the entity MUST have numerically smaller `version` values
+  - MUST be unique across all versions of the entity
+- Examples:
+  - `1`, `2`, `3`
+
+### `epoch`
+- Type: Unisgned Integer
+- Description: A numeric value used to determine whether an entity has been
+  modified. Each time the associated entity is updated, this value MUST be
+  set to a new value that is greater than the current one.
+  Note that unlike `version`, this attribute is most often managed by
+  the Registry itself. Additionally, if an entity is created that is based
+  on another entity (e.g. a new version of an entity is created), then the
+  new entity's `epoch` value can be reset (e.g. to zero) since the scope of
+  its values is the entity.
+- Constraints:
+  - MUST be an unsigned integer equal to or greater than zero
+  - MUST be unique within the scope of a version of an entity. If the entity
+    is not versioned then the scope is just the entity itself
+- Examples:
+  - `1`, `2`, `3`
+
+### `self`
+- Type: URL
+- Description: A unique URL for the entity. The URL MUST be a combination of
+  the base URL for the list of resources of this type of entity appended with
+  the `id` of the entity.
+- Constraints:
+  - MUST be a non-empty URL
+- Examples:
+  - `https://example.com/registry/endpoints/123`
+
+### `createdBy`
+- Type: String
+- Description: A reference to the user or component that was responsible for
+  the creation of this entity. This specification makes no requirement on
+  the semantics or syntax of this value.
+- Constraints:
+  - When this attributes has no value it MUST be serialized by either an empty
+    string or by being excluded from the serialization of the owning entity
+- Examples:
+  - `John Smith`
+  - `john.smith@example.com`
+
+### `createdOn`
+- Type: Timestamp
+- Description: The date/time of when the entity was created.
+- Constraints:
+  - MUST be an [RFC3339](https://tools.ietf.org/html/rfc3339) timestamp
+- Examples:
+  - `2030-12-19T06:00:00Z"
```

Comment:
nit: Inconsistent use of ' and "

Reply:
LOL took me a while to realize you wrote "and" not "an". I just couldn't find "and" in this section!
Thanks for noticing my copy-n-paste error.